README.md

```diff
@@ -1 +1,810 @@
+- A Text Editor or an IDE 
+
+## Implementation
+
+> If you want to skip the basics, you can download the GitHub repo and continue from the "Testing" section.
+
+### Implementing database
+ - Start MYSQL server in your local machine.
+ - Create a database with name `testdb` in your MYSQL localhost. If you want to skip the database implementation, then directly import the [testdb.sql](https://github.com/Shairam/ballerina-sp-tracing/blob/master/resources/testdb.sql) file into your localhost. You can find it in the Github repo.
+ 
+ 
+ 
+
+### Create the project structure
+        
+ For the purpose of this guide, let's use the following package structure.
+        
+    
+    ballerina-sp-tracing
+           └── guide
+                ├── students
+                │   ├── student_management_service.bal
+                │   ├── student_marks_management_service.bal
+                ├── client_service
+                |         └── client_main.bal
+                └── ballerina.conf
+        
+
+- Create the above directories in your local machine, along with the empty `.bal` files.
+
+- You have to add the following lines in your [ballerina.conf](https://github.com/Shairam/ballerina-sp-tracing/blob/master/ballerina.conf).
+
+```ballerina
+[b7a.observability.tracing]
+enabled=true
+name="wso2sp"
+
+[b7a.observability.tracing.wso2sp]
+reporter.wso2sp.publisher.type="thrift"
+reporter.wso2sp.publisher.username="admin"
+reporter.wso2sp.publisher.password="admin"
+reporter.wso2sp.publisher.url="tcp://localhost:7611"
+reporter.wso2sp.publisher.authUrl="ssl://localhost:7711"
+reporter.wso2sp.publisher.databridge.agent.config="<SET ABSOLUTE PATH>/data.agent.config.yaml"
+javax.net.ssl.trustStore="<SET ABSOLUTE PATH>/wso2carbon.jks"
+javax.net.ssl.trustStorePassword="admin"
+reporter.wso2sp.publisher.service.name="ballerina_hello_world"
+```
+- In the ballerina.conf file for line number 11 and 12, you are required to give the appropriate absolute path to the following files mentioned in the lines.
+- You can find these files [here](https://github.com/Shairam/ballerina-sp-tracing/tree/extras-1/resources/main/resources).
+- Also you need to update the [data.agent.config.yaml](https://github.com/Shairam/ballerina-sp-tracing/blob/extras-1/resources/main/resources/data.agent.config.yaml) file by including the absolute path of the [required files](https://github.com/Shairam/ballerina-sp-tracing/tree/extras-1/resources/main/resources) in the following fields. 
+  - trustStorePath, keystoreLocation, secretPropertiesFile, masterKeyReaderFile .
+- Then open the terminal and navigate to `ballerina-sp-tracing/guide` and run Ballerina project initializing toolkit.
+
+  ``
+     $ ballerina init
+  ``
+  
+  
+- Also you need to clone and build the ballerina-sp-extension in the following repository [https://github.com/ballerina-platform/ballerina-observability](https://github.com/ballerina-platform/ballerina-observability) 
```

Comment:
Shall we give the link to the sp-extension in this repo?

Reply:
Noted. I will update it.